DuckDuckGo/UserBehaviorMonitor.swift

```diff
@@ -81,22 +81,33 @@ final class UserBehaviorMonitor {
         set { store.didRefreshCounter = newValue }
     }
 
-    func handleRefreshAction(date: Date = Date()) {
-        fireEventIfActionOccurredRecently(within: 12.0, since: didRefreshTimestamp, eventToFire: .reloadTwiceWithin12Seconds)
-        didRefreshTimestamp = date
-
-        if didRefreshCounter == 0 {
-            didDoubleRefreshTimestamp = date
-        }
-        didRefreshCounter += 1
-        if didRefreshCounter > 2 {
-            fireEventIfActionOccurredRecently(within: 20.0, since: didDoubleRefreshTimestamp, eventToFire: .reloadThreeTimesWithin20Seconds)
-            didRefreshCounter = 0
+    func handleAction(_ action: Action, date: Date = Date()) {
+        switch action {
+        case .refresh:
+            fireEventIfActionOccurredRecently(within: 12.0, since: didRefreshTimestamp, eventToFire: .reloadTwiceWithin12Seconds)
+            fireEventIfActionOccurredRecently(within: 24.0, since: didRefreshTimestamp, eventToFire: .reloadTwiceWithin24Seconds)
+            didRefreshTimestamp = date
+            
+            if didRefreshCounter == 0 {
+                didDoubleRefreshTimestamp = date
+            }
+            didRefreshCounter += 1
+            if didRefreshCounter > 2 {
+                fireEventIfActionOccurredRecently(within: 20.0, since: didDoubleRefreshTimestamp, eventToFire: .reloadThreeTimesWithin20Seconds)
+                fireEventIfActionOccurredRecently(within: 40.0, since: didDoubleRefreshTimestamp, eventToFire: .reloadThreeTimesWithin40Seconds)
+                didRefreshCounter = 0
+            }
+        case .reopenApp:
+            fireEventIfActionOccurredRecently(within: 30.0, since: didRefreshTimestamp, eventToFire: .reloadAndRestartWithin30Seconds)
+            fireEventIfActionOccurredRecently(within: 50.0, since: didRefreshTimestamp, eventToFire: .reloadAndRestartWithin50Seconds)
```

Comment:
Please change selected lines in this comment to what it was, as we only want to support reloadTwiceWithin12Seconds and reloadThreeTimesWithin20Seconds user actions